src/main/java/com/google/swarm/tokenization/common/DLPTokenizationDoFn.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.beam.sdk.values.KV;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.cloud.dlp.v2.DlpServiceClient;
+import com.google.privacy.dlp.v2.ContentItem;
+import com.google.privacy.dlp.v2.DeidentifyContentRequest;
+import com.google.privacy.dlp.v2.DeidentifyContentResponse;
+import com.google.privacy.dlp.v2.FieldId;
+import com.google.privacy.dlp.v2.ProjectName;
+import com.google.privacy.dlp.v2.Table;
+
+public class DLPTokenizationDoFn extends DoFn<KV<String, Table>, Row> {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -8845993157503721016L;
+	public static final Logger LOG = LoggerFactory.getLogger(DLPTokenizationDoFn.class);
+	private String projectId;
+	private DlpServiceClient dlpServiceClient;
+	private ValueProvider<String> deIdentifyTemplateName;
+	private ValueProvider<String> inspectTemplateName;
+	private boolean inspectTemplateExist;
+
+	public DLPTokenizationDoFn(String projectId, ValueProvider<String> deIdentifyTemplateName,
+			ValueProvider<String> inspectTemplateName) {
+		this.projectId = projectId;
+		dlpServiceClient = null;
+		this.deIdentifyTemplateName = deIdentifyTemplateName;
+		this.inspectTemplateName = inspectTemplateName;
+		inspectTemplateExist = false;
+
+	}
+
+	@StartBundle
+	public void startBundle() throws SQLException {
+
+		try {
+			this.dlpServiceClient = DlpServiceClient.create();
+		} catch (IOException e) {
+
+			e.printStackTrace();
+		}
+
+	}
+
+	@FinishBundle
+	public void finishBundle() throws Exception {
+		if (this.dlpServiceClient != null) {
+			this.dlpServiceClient.close();
+		}
+	}
+
+	@ProcessElement
+	public void processElement(ProcessContext c) {
+
+		String key = c.element().getKey();
+		Table nonEncryptedData = c.element().getValue();
+		if (this.inspectTemplateName.isAccessible()) {
+			if (this.inspectTemplateName.get() != null)
+				this.inspectTemplateExist = true;
+		}
+		ContentItem tableItem = ContentItem.newBuilder().setTable(nonEncryptedData).build();
+		DeidentifyContentRequest request;
+		DeidentifyContentResponse response;
+		if (this.inspectTemplateExist) {
+			request = DeidentifyContentRequest.newBuilder().setParent(ProjectName.of(this.projectId).toString())
+					.setDeidentifyTemplateName(this.deIdentifyTemplateName.get())
+					.setInspectTemplateName(this.inspectTemplateName.get()).setItem(tableItem).build();
+		} else {
+			request = DeidentifyContentRequest.newBuilder().setParent(ProjectName.of(this.projectId).toString())
+					.setDeidentifyTemplateName(this.deIdentifyTemplateName.get()).setItem(tableItem).build();
+
+		}
+
+		response = dlpServiceClient.deidentifyContent(request);
+		Table encryptedData = response.getItem().getTable();
+		LOG.info("Request Size Successfully Tokenized:{} rows {} bytes ", encryptedData.getRowsList().size(),
+				request.toByteString().size());
+
+		List<String> outputHeaders = encryptedData.getHeadersList().stream().map(FieldId::getName)
```

Comment:
Should we put this `get headers list/check` in a method?

Reply:
@theodoresiu getHeaderList is part of DLP table API. Do you mean add some validation to it before joining?